op-challenger/game/fault/agent_test.go

```diff
@@ -127,20 +127,31 @@ func TestAgent_SelectiveClaimResolution(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
-		test := test
-		t.Run(test.name, func(t *testing.T) {
+	for _, tCase := range tests {
+		tCase := tCase
+		t.Run(tCase.name, func(t *testing.T) {
 			agent, claimLoader, responder := setupTestAgent(t)
-			agent.selective = test.selective
-			agent.claimants = test.claimants
+			agent.selective = tCase.selective
+			agent.claimants = tCase.claimants
 			claimLoader.maxLoads = 1
-			claimLoader.claims = test.claims
-			responder.callResolveStatus = test.callResolveStatus
+			if tCase.selective {
+				claimLoader.maxLoads = 0
+			}
+			claimLoader.claims = tCase.claims
+			responder.callResolveStatus = tCase.callResolveStatus
+			responder.onResolveClaims = func(claims ...uint64) {
+				// Ensure there are no more claims for us to resolve, otherwise we loop forever trying to resolve them
+				claimLoader.claims = []types.Claim{test.NewClaimBuilder(t, 4, alphabet.NewTraceProvider(big.NewInt(0), 4)).CreateRootClaim()}
```

Comment:
I don't see why you needed to add this new logic here.

Reply:
Because the attempt to resolve claims runs in a loop until there are no claims to resolve (which it detects by getting an error back). Previously the error it got back was when it tried to reload the claims after the first call to resolveClaim and it would then try to resolve the game, get a status that isn't InProgress and skip trying to perform any moves.  Now that it doesn't try to resolve the game, it doesn't see the game as resolved and does try to perform moves - the first thing that does is load claims so we can't have that return an error anymore.  And when that doesn't return an error the `resolveClaim` loop keeps retrying forever because it always sees a claim as resolvable (since the test wasn't ever actually resolving it).

This logic means that when the claim is resolved it gets removed from the game which isn't exactly accurate but gets us to the point we need where it doesn't see any more resolvable claims and can still run through the perform move logic like it should.

Reply:
The infinite loop was confusing me.  But I guess when this actually runs, it will resolve what it can.  And then on the next round of checks, the resolved claims are no longer resolvable (because they're already resolved), so it moves on?  Would be nice if the tests mimicked the expected behavior more closely.

Reply:
Yeah exactly.  I'll take a look and see if I can make it more accurate - might be able to record which claims are resolved in the stub and make it work.